lib/util/propTypes.js

```diff
@@ -281,6 +281,44 @@ module.exports = function propTypesInstructions(context, components, utils) {
     return ignorePropsValidation;
   }
 
+  function declarePropTypesForTSTypeAnnotation(propTypes, declaredPropTypes) {
+    let members = [];
+
+    if (propTypes.typeAnnotation.type === 'TSTypeReference') {
+      const typeName = propTypes.typeAnnotation.typeName.name;
+      if (!typeName) {
+        return true;
+      }
+      // the game here is to find the type declaration in the code
+      const candidateTypes = context.getSourceCode().ast.body.filter((item) => item.type === 'VariableDeclaration' && item.kind === 'type');
+      const declarations = candidateTypes.map((type) => type.declarations).reduce((acc, val) => acc.concat(val), []);
+      const type = declarations.find((dec) => dec.id.name === typeName);
+
+      if (!type) {
+        return true;
+      }
+
+      members = type.init.members;
+    } else if (propTypes.typeAnnotation.type === 'TSTypeLiteral') {
+      members = propTypes.typeAnnotation.members;
+    } else {
+      return true; // we don't handle cases where the type is declared directly. Can be improved in the future, with TSTypeLiteral for example
```

Comment:
What would be examples of this?

Reply:
see above

Reply:
hmm. so the choice is between:
1. possibly give a false positive, which would force the user to change their code how?
1. silently accept something that should be warned on
?

Reply:
I think both cases are weird, but I have a slight preference for 1 :
1. setting ignore at true : what happens in this case ? does it emit a warning ? Maybe in the future we could signal the user that the props are not checked because of improper declaration/compatibility issue with the rule ? It could be an option, like "props validation has been ignored for this component" no ?
2. declaring props as {} (ignore = false, members = []): in this case, it fails silently since there is no unused props in {}. 

I think 1 gives a bit more perspective for this rule, as 2 just swallows the error and the next maintainer will have to dig in the code to understand why declared props are {} (as I had to do to fix the issue here)

what do you think :) ?

Reply:
seems like the first option is better

Reply:
ok i leave it as it is now then 😃